Module: add dump function that recursively prints contents of the module.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24356 Module: add dump function that recursively prints contents of the module.**
* #24355 jit_log: Extract a function that prefixes all lines of a string with another string.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24356

Differential Revision: [D16864133](https://our.internmc.facebook.com/intern/diff/D16864133)

Commit:
Module: add debug_print function that recursively prints contents of the module.

Commit:
Update on "Module: add debug_print function that recursively prints contents of the module."

suo commented on torch/csrc/jit/script/module.h:
```diff
@@ -205,6 +205,11 @@ struct TORCH_API Module {
   slot_list get_attributes() const;
   slot_list get_module_slots() const;
 
+  void debug_print(
```
can we call it `dump` instead, to better align with similar names elsehwere?

suo reviewed (approved):
seems legit

Commit:
Rename 'debug_print' to 'dump' on "Module: add dump function that recursively prints contents of the module."


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24356

Differential Revision: [D16824511](https://our.internmc.facebook.com/intern/diff/D16824511)
